nevergrad/optimization/optimizerlib.py

```diff
@@ -517,6 +517,45 @@ def _internal_tell_not_asked(self, candidate: base.Candidate, value: float) -> N
         self._internal_tell_candidate(candidate, value)  # go through standard pipeline
 
 
+
+@registry.register
+class CTBPSA(base.Optimizer):
+    def __init__(self, instrumentation: Union[int, Instrumentation], budget: Optional[int] = None, num_workers: int = 1) -> None:
+        super().__init__(instrumentation, budget=budget, num_workers=num_workers)
+        dimension = self.dimension
+        self.tbpsa = TBPSA(dimension, budget=budget, num_workers=num_workers)
+        self.num_repeat = 0
+        self.never_decrease = True
+        self.repetitions: Dict[bytes, Tuple[int, List[float]]] = {}
+        self.current_point: np.ndarray = np.zeros(dimension)
+
+    def _internal_ask(self) -> base.ArrayLike:
+        if self.num_repeat == 0:
+            self.current_point = self.tbpsa.ask().data  #, copy=False)
+            xbytes = np.array(self.current_point, copy=False).tobytes()
+            self.num_repeat = max(1, int(self.tbpsa.llambda**.1))
+            self.repetitions[xbytes] = (self.num_repeat, [])
+        self.num_repeat -= 1
+        return self.current_point
+
+    def _internal_tell(self, x: base.ArrayLike, value: float) -> None:
+        x = np.array(x, copy=False)
+        xbytes = x.tobytes()
+        total, values = self.repetitions[xbytes]
+        values.append(value)
+        if len(values) == total:
+            self.tbpsa._internal_tell(x, np.mean(values))
+            del self.repetitions[xbytes]
+
+    def _internal_provide_recommendation(self) -> base.ArrayLike:  # This is NOT the naive version. We deal with noise.
+        return self.tbpsa.provide_recommendation().data
+
+    def tell_not_asked(self, x: base.ArrayLike, value: float) -> None:
+        raise base.TellNotAskedNotSupportedError
+
+    def _internal_tell_not_asked(self, candidate: base.Candidate, value: float) -> None:
```

Comment:
Since it is based on `TBPSA` which supports tell not asked, this could be made to work